server/src/main/java/org/elasticsearch/monitor/jvm/SunThreadInfo.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.monitor.jvm;
+
+import com.sun.management.ThreadMXBean;
+import java.lang.management.ManagementFactory;
+
+public class SunThreadInfo {
+
+    private static final ThreadMXBean threadMXBean;
+
+    static {
+        threadMXBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
```

Comment:
I asked for my JDK team in my company, they give me an other way (reflection) to avoid import com.sun package, i think both approaches are good

````
import java.lang.management.ManagementFactory;
import java.lang.reflect.Method;

import java.lang.management.ThreadMXBean;

public class Test {
    public static void main(String[] args) {
        ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean(); 
        try { 
            Method getBytes = threadMXBean.getClass().getMethod("getThreadAllocatedBytes", long.class); 
            getBytes.setAccessible(true); 

            long threadId = Thread.currentThread().getId();
            long bytes = (long)getBytes.invoke(threadMXBean, threadId);
            System.out.println(bytes);
        } catch (Throwable e) { 
            System.out.println(e);
        }
    }
}
````

Reply:
Both approaches are good, we import the com.sun packages in a few places already in Elasticsearch, so it's not unusual. We actually do something similar to the above (with reflection) to get access to the HotSpotDiagnosticMXBean class in JvmInfo.java here: (https://github.com/elastic/elasticsearch/blob/a2373c3043adc11e79664d80865985635077f7be/server/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java#L94).

You can borrow some code from there if you'd like, HotSpotDiagnosticMXBean is in the same package as com.sun.management.ThreadMXBean.